org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/BaseDiagnosticsHandler.java

```diff
@@ -127,7 +131,27 @@ public void beginReporting() {
 	public void endReporting() {
 		JavaLanguageServerPlugin.logInfo(problems.size() + " problems reported for " + this.uri.substring(this.uri.lastIndexOf('/')));
 		boolean isDiagnosticTagSupported = JavaLanguageServerPlugin.getPreferencesManager().getClientPreferences().isDiagnosticTagSupported();
-		PublishDiagnosticsParams $ = new PublishDiagnosticsParams(ResourceUtils.toClientUri(uri), toDiagnosticsArray(this.cu, problems, isDiagnosticTagSupported));
+		List<Diagnostic> diagnostics = toDiagnosticsArray(this.cu, problems, isDiagnosticTagSupported);
+		try {
+			List<IMarker> list = new ArrayList<>();
+			if (cu != null && cu.getUnderlyingResource() != null) {
+				IMarker[] markers = cu.getUnderlyingResource().findMarkers(null, true, IResource.DEPTH_ONE);
+				for (IMarker marker : markers) {
+					if (IJavaModelMarker.JAVA_MODEL_PROBLEM_MARKER.equals(marker.getType()) || IJavaModelMarker.TASK_MARKER.equals(marker.getType())) {
+						continue;
+					}
+					list.add(marker);
+				}
+			}

```

Comment:
you should add a comment to explain what is happening and why

Reply:
or extract this snippet to a `collectNonJavaProblems(cu)` method

Reply:
i.e. from L135 to L153

Reply:
Fixed.